### PR TITLE
[radio] align RAIL TX buffer to appropriate memory alignment

### PR DESCRIPTION
**Pull request by lmnotran**
This aligns the RAIL TX buffer to the appropriate memory alignment for each specific series of chips.

### CODE REVIEW

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by romacdon — APPROVED** [review]
LGTM

**Comment by jwhui** [comment]
Are we still waiting on @schinchani 's review?

**Comment by lmnotran** [comment]
Yes, please wait for his review, but I also need to check one last thing. I'll ping you when this is ready to merge @jwhui. Thanks for checking!

**Review by schinchani — APPROVED** [review]
LGTM.

**Comment by lmnotran** [comment]
@jwhui I just finished doing some final checks. Please merge this PR when you get the chance. Thanks! :)

**Review by jwhui — APPROVED** [review]

**Pull request merged**
